### PR TITLE
Replace Sequential sampler to Batch + PL trainer cfg_parser

### PR DESCRIPTION
**Pull request by DaloroAT**

### CODE REVIEW

File: oml/lightning/entrypoints/utils.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    There are two parameters 'accelerator' and 'devices' you can configure. If one of the parameters or both of them
+    are not specified, the best option based on your hardware will be automatically prepared for you.
+    Posible options for 'accelerator' are 'cpu' and 'gpu'.
+    You can select specific gpus using list 'devices=[0, 2]' or specify the number of gpus by `devices=3`.
+    Experiment might be launched in DDP with 'cpu' accelerator. In this case, 'devices' (integer value or len of list)
+    interpreted as number of processes.
+
+    TODO: now DDP works in wrong way. We temporarily force to specify 'devices=1'
+    """
+    cfg = dictconfig_to_dict(cfg.copy())
```
Comment (b0nce) [inline oml/lightning/entrypoints/utils.py:22]: why do we need to pass `cfg.copy()`?

Comment (DaloroAT) [reply oml/lightning/entrypoints/utils.py]: rmed

File: oml/interfaces/samplers.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+from abc import ABC, abstractmethod
+from typing import Iterator, List
+
+
+class IBatchSampler(ABC):
+    """
+    We introduce our own IBatchSampler interface instead of using default torch BatchSampler, because the last on is
```
Comment (AlekseySh) [inline oml/interfaces/samplers.py:7]: last one*

File: oml/lightning/entrypoints/parser.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
+from typing import Any, Dict
+
+import torch
+from pytorch_lightning.plugins import DDPPlugin
+
+from oml.const import TCfg
+from oml.utils.misc import dictconfig_to_dict
+
+
+def parse_runtime_params_from_config(cfg: TCfg) -> Dict[str, Any]:
```
Comment (AlekseySh) [inline oml/lightning/entrypoints/parser.py:10]: rename to `parse_engine_params_from_config` ?

File: oml/lightning/entrypoints/parser.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    """
+    Function parses cfg and based on it prepares DDP parameters for PytorchLightning Trainer module.
+    There are two parameters 'accelerator' and 'devices' you can configure. If one of the parameters or both of them
+    are not specified, the best option based on your hardware will be automatically prepared for you.
+    Posible options for 'accelerator' are 'cpu' and 'gpu'.
+    You can select specific gpus using list 'devices=[0, 2]' or specify the number of gpus by `devices=3`.
+    Experiment might be launched in DDP with 'cpu' accelerator. In this case, 'devices' (integer value or len of list)
+    interpreted as number of processes.
+
+    TODO: now DDP works in wrong way. We temporarily force to specify 'devices=1'
```
Comment (AlekseySh) [inline oml/lightning/entrypoints/parser.py:20]: one todo below is enough

File: oml/lightning/entrypoints/parser.py [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    if isinstance(devices, (list, tuple)) and accelerator == "cpu":
+        devices = len(devices)
+
+    if (isinstance(devices, int) and devices > 1) or (isinstance(devices, (list, tuple)) and len(devices) > 1):
+        strategy = DDPPlugin()
+    else:
+        strategy = None
+
+    # TODO: we will remove it after merging PR with proper DDP
+    if not (strategy is None):
```
Comment (AlekseySh) [inline oml/lightning/entrypoints/parser.py:42]: if strategy:

File: oml/lightning/entrypoints/parser.py (hunk 3 above)
Comment (AlekseySh) [inline oml/lightning/entrypoints/parser.py:10]: we probably need some tests for this parser

File: oml/lightning/entrypoints/train.py [hunk 6]
```diff
@@ -105,14 +108,21 @@ def pl_train(cfg: TCfg) -> None:
...
-        sampler=sampler,
-        num_workers=cfg["num_workers"],
-        batch_size=sampler.batch_size,
-        drop_last=True,
-        shuffle=not bool(sampler),
-    )
+    if sampler is None:
+        loader_train = DataLoader(
+            dataset=train_dataset,
+            sampler=sampler,
```
Comment (AlekseySh) [inline oml/lightning/entrypoints/train.py:114]: rm this row

File: tests/test_examples/test_runs_via_configs/configs/train_with_bboxes.yaml [hunk 7]
```diff
@@ -3,6 +3,8 @@ postfix: "deepfashion"
 seed: 42
 gpus: []
 precision: 32
+accelerator: cpu
+devices: 1
```
Comment (AlekseySh) [inline tests/test_examples/test_runs_via_configs/configs/train_with_bboxes.yaml:7]: let's use devices > 1 at least in one of the mock runs

**Review by AlekseySh — CHANGES_REQUESTED** [review]
please, check spelling in docs with Grammarly (for example, there are some missed articles)

**Pull request merged**